### DISCUSSION TITLE
Question about .ppm generation code

### DISCUSSION THREAD

**Discussion by hbhutta** [RayTracing/raytracing.github.io#1107] [Q&A] [answered]
I've just started going through the In a Weekend book and I've been playing around with the code to generate an image as a .ppm file.
I've been able to make a vertical blue into white gradient by changing the values in the code a bit. I am confused about the first line in the `for` loop. I understand that the origin is in the top-left corner. So why do we decrement `j` from the bottom (the value of the height), and bring it towards the top? Does it make a difference as long as we color all the pixels in the image (I understand that there will be `WIDTH * HEIGHT` number of pixels)?

![image](https://user-images.githubusercontent.com/112280419/234952405-5b0599f5-8096-4633-8596-57d76c080e84.png)

**Comment by hollasch** [comment, marked as answer]
It's just a test image, so there's really no true top or bottom. In this example, we're going from no green at the top to full green at the bottom. We set things up this way because as you progress to actual images, there's a definite top to the image. In this case, we're going from a 3D coordinate system where Y points up, to an image coordinate system where 0 denotes the top line and we increase as we go down. So while we don't really need to worry about the inversion at this precise point, we definitely will need to in the future.
